lib/rspec/core/formatters/snippet_extractor.rb

```diff
@@ -1,114 +1,133 @@
+RSpec::Support.require_rspec_core "source"
+
 module RSpec
   module Core
     module Formatters
-      # @api private
-      #
-      # Extracts code snippets by looking at the backtrace of the passed error
-      # and applies synax highlighting and line numbers using html.
+      # @private
       class SnippetExtractor
```

Comment:
The new `SnippetExtractor` is generic and not limited for terminal use. Maybe it can be used even in `HtmlSnippetExtractor` for the future.